### PR TITLE
[NP-2436] Add ReviewCapability

### PR DESCRIPTION
**Pull request by KernelDeimos**
A ReviewCapability is a Capability which asks a user to review another Capability they previously obtained. It has a corresponding Wizardlet which displays the capability being reviewed, provides an `Edit` button which opens this capability in a new wizard, and prevents continuing until the review checkbox is checked.

TODO: it may be desired to omit this review checkbox and have the Submit button read "Accept" instead. This will require some additional work to allow wizardlets to determine the labels on the wizard. A caveat to this would be that the user might not scroll to the bottom before hitting accept.

Screen Shot 2020-10-26 at 2 13 50 AM

### CODE REVIEW

File: src/foam/nanos/crunch/ui/UCJView.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          }
+          this.ucjData = defaultData;
+        })
+      });
+    }
+  ],
+
+  actions: [
+    function edit() {
+      console.log('actionThis', this);
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/ui/UCJView.js]: remove

File: src/foam/nanos/crunch/ui/UCJView.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        .add(this.CreateWizardletsAgent)
+        .add(this.LoadTopConfig)
+        .add(this.StepWizardAgent)
+        .add(this.PutFinalJunctionsAgent)
+        .execute()
+        .then(() => {
+          this.update();
+        })
+        .catch(err => {
+          debugger;
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/ui/UCJView.js]: remove

File: src/foam/nanos/crunch/extra/ReviewCapability.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  properties: [
+    {
+      class: 'Object',
+      name: 'wizardlet',
+      documentation: `
+        Defines a wizardlet used when displaying this capability on related client crunch wizards.
+      `,
+      factory: function() {
+        return foam.nanos.crunch.ui.ReviewCapabilityWizardlet.create({}, this);
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/extra/ReviewCapability.js]: Better to require: this class then do this.ReviewCapabilityWizardlet() instead. Helps the build follow dependencies better.

File: src/foam/nanos/crunch/extra/ReviewCapability.js [hunk 4]
```diff
@@ -21,7 +25,7 @@ foam.CLASS({
         Defines a wizardlet used when displaying this capability on related client crunch wizards.
       `,
       factory: function() {
-        return foam.nanos.crunch.ui.ReviewCapabilityWizardlet.create({}, this);
+        return this.ReviewCapabilityWizardlet.create({}, this);
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/extra/ReviewCapability.js]: You don't need "{}, this" anymore.

Comment (KernelDeimos) [reply src/foam/nanos/crunch/extra/ReviewCapability.js]: Oh nice!

File: src/foam/nanos/crunch/extra/ReviewCapabilityData.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.extra',
+  name: 'ReviewCapabilityData',
+
+  messages: [
+    { name: 'REVIEW_REQUIRED_ERROR', message: 'All data must be reviewed by a signing officer.' }
```
Comment (jlhughes) [inline src/foam/nanos/crunch/extra/ReviewCapabilityData.js]:
```suggestion
    { name: 'REVIEW_REQUIRED_ERROR', message: 'All data must be reviewed by a Signing Officer.' }
```

Comment (KernelDeimos) [reply src/foam/nanos/crunch/extra/ReviewCapabilityData.js]: Changed to `privileged user`

**Pull request merged**